src/generate.cpp

```diff
@@ -6,32 +6,81 @@ inline namespace MIGRAPHX_INLINE_NS {
 argument fill_argument(shape s, unsigned long value)
 {
     argument result;
-    s.visit_type([&](auto as) {
-        using type = typename decltype(as)::type;
-        auto v     = fill_tensor_data<type>(s, value);
-        result     = {s, v};
-    });
+    if(s.type() == shape::tuple_type)
+    {
+        std::vector<argument> sub_args;
+        const auto& sub_ss = s.sub_shapes();
+        std::transform(sub_ss.begin(), sub_ss.end(), std::back_inserter(sub_args), [&](auto ss) {
+            argument temp_arg;
+            ss.visit_type([&](auto as) {
+                using type = typename decltype(as)::type;
+                auto v     = fill_tensor_data<type>(ss, value);
+                temp_arg   = {ss, v};
+            });
+
+            return temp_arg;
+        });
```

Comment:
This should call generate recursively because if this is a tuple within a tuple it will throw an error.

Reply:
fixed